core/tests/puppeteer-acceptance-tests/utilities/user/logged-in-user.ts

```diff
@@ -2817,20 +2824,44 @@ export class LoggedInUser extends BaseUser {
   /**
    * Checks if Learner is on the learner dashboard page.
    */
-  expectToBeOnLearnerDashboardPage(): void {
-    expect(this.page.url()).toBe(`${baseUrl}/learner-dashboard`);
+  async expectToBeOnLearnerDashboardPage(): Promise<void> {
+    await this.page.waitForFunction(
+      (url: string) => {
+        return window.location.href.includes(url);
+      },
+      {},
+      '/learner-dashboard'
+    );
   }
 
   /**
-   * Checks if greeting has name of the user
+   * Checks if greeting has name of the user.
    */
   async expectGreetingToHaveNameOfUser(userName: string): Promise<void> {
-    const greetingElement = await this.page.$(greetingSelector);
-    const greetingText = await this.page.evaluate(
-      el => el.textContent,
-      greetingElement
+    // Check for redesigned dashboard greeting first.
+    const redesignedGreetingElement = await this.page.$(
+      learnerGreetingsSelector
     );
```

Comment:
In a case where the redesigned dashboard is enabled, but elements hasn't loaded yet, it generates a flaky behaviour. You can use `isElementVisible` that waits sufficiently to remove flakiness.